src/event/event.c

```diff
@@ -792,7 +792,7 @@ static void
 _dispatch_timer_unote_disarm(dispatch_timer_source_refs_t dt,
 		dispatch_timer_heap_t dth)
 {
-	uint32_t tidx = dt->du_ident;
+	uint32_t tidx = (uint32_t)dt->du_ident;
```

Comment:
Do we need to worry about truncation?  IIRC, a timer on Windows would be a `HANDLE` which would be truncated by `uint32_t` as it should be `uintptr_t`.

Reply:
`du_ident` on Windows is indeed a `uintptr_t`, as it is on OpenBSD. The timer index here is a `uint32_t`, so as written there is already a truncation occurring. `du_ident` is already initialized in various places from `_dispatch_timer_unote_idx` which returns an `unsigned int`, but here we access `du_ident` directly, which is not guaranteed to be 32-bit wide. 

Casting here just suppresses the error. I don't think there is actually a problem here on looking how du_ident is used in this module; the 32-bit assumption is baked a little hard in the timer code and disentangling that should be handled separately.